salt/orch/update-etc-hosts.sls

```diff
@@ -1,13 +1,18 @@
+{% if salt['mine.get']('roles:kube-(master|minion)', 'caasp_fqdn', expr_form='grain_pcre').values()|length > 0 %}
 update_pillar:
   salt.function:
-    - tgt: '*'
+    - tgt: 'roles:kube-(master|minion)'
+    - tgt_type: grain_pcre
     - name: saltutil.refresh_pillar
 
 update_grains:
   salt.function:
-    - tgt: '*'
+    - tgt: 'roles:kube-(master|minion)'
```

Comment:
It is a little more complex. 

We do not care if the IP has changed before the highstate has run. After the highstate has run, we care.

e.g for workers - if an IP changes before https://github.com/kubic-project/salt/blob/master/salt/orch/kubernetes.sls#L110 - we don't really care as the highstate should deal with it.